CommonTools/RecoAlgos/plugins/JetConstituentSelector.cc

```diff
@@ -28,76 +28,55 @@
 #include "CommonTools/UtilAlgos/interface/StringCutObjectSelector.h"
 #include "FWCore/Framework/interface/Event.h"
 
-template < class T, typename C = std::vector<typename T::ConstituentTypeFwdPtr> >
-class JetConstituentSelector : public edm::EDFilter {
-
+template <class T, typename C = std::vector<typename T::ConstituentTypeFwdPtr>>
+class JetConstituentSelector : public edm::stream::EDFilter<> {
```

Comment:
Sounds reasonable, although it requires changing the following configuration files:
```
RecoJets/Configuration/python/RecoPFJets_cff.py:ak8PFJetsCHSConstituents = cms.EDFilter("PFJetConstituentSelector",
RecoJets/JetProducers/python/ak8PFJetsCS_cfi.py:ak8PFJetsCSConstituents = cms.EDFilter("PFJetConstituentSelector",
```